### DISCUSSION TITLE
Optionally force default locale url prefix

### DISCUSSION THREAD

**Discussion by 11koukou** [vercel/next.js#38389] [Ideas]
### Describe the feature you'd like to request

As per the #18419 discussion I would like to open a feature request just in case that the mentioned post is no longer being monitored due to having being "answered". 

Well, it is a much requested feature to optionally force the locale appear in the url even if it is the default one. The current middleware workaround is more or less what its name exactly means: a workaround. It introduces  unnecessary complexity and even drops performance, for something that is hypothetically simple.

### Describe the solution you'd like

Having a "forceDefaultLocalePrefix" boolean flag in next.config.js

### Describe alternatives you've considered

None

**Comment by SmoshySmosh** [comment]
This is a 100% necessary feature that Next.js should support. 

If you use NX like I do, the "default" hack will prevent builds from being successful due to failures getting the `default` translations during the build. One way around this is creating a `public/locales/default/common.json` file, but honestly this seems completely unnecessary. This may impact also Next.js without NX, but I haven't seen similar issues in the comments and I haven't tested it myself.

The behavior without the `default` hack is inconsistent as well.
`/` -> `/`: Not expected, should redirect to the default locale of the browser or fallback to `/en-us` if the browser locale is not in the list of supported locales
`/en-us` -> `/`: Not expected, should stay `/en-us`
`/es-es` -> `/es-es`: Expected, stays on `/es-es`

If I implement middleware logic to redirect to my default locale, `/en-us`, I will get stuck in an infinite redirect loop. I'm open to suggestions here, but it seems like the documented path forward on this does not yield the expected results.
